src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -28,66 +29,65 @@ public Recoverability()
                 settings.SetDefault(SlrTimeIncrease, DefaultSecondLevelRetryPolicy.DefaultTimeIncrease);
 
                 settings.SetDefault(FlrNumberOfRetries, 5);
-
-                settings.SetDefault(FailureInfoStorageCacheSizeKey, 1000);
             });
         }
+
         protected internal override void Setup(FeatureConfigurationContext context)
         {
             var errorQueue = context.Settings.ErrorQueueAddress();
             context.Settings.Get<QueueBindings>().BindSending(errorQueue);
 
-            var failureInfoStorage = new FailureInfoStorage(context.Settings.Get<int>(FailureInfoStorageCacheSizeKey));
-            var localAddress = context.Settings.LocalAddress();
+            var transportTransactionMode = context.Settings.GetRequiredTransactionModeForReceives();
+
+            if (transportTransactionMode == TransportTransactionMode.None)
+            {
+                Logger.Warn("First and Second Level Retries will be disabled. Automatic retries are not supported when running with TransportTransactionMode.None. Failed messages will be moved to the error queue instead.");
```

Comment:
should we only warn in case FLR/SLR are actually configured to be enabled (e.g. do not log a warning if a user disabled them?)